### PR TITLE
Enable all tests on 3.13 for Linux and macOS

### LINKED ISSUE #12060: util.getPassword fix for Python 3.13
```
Traceback (most recent call last):
  File "/home/adi/chevah/twisted/src/twisted/python/test/test_util.py", line 462, in processFinished
    reason.trap(ProcessDone)
  File "/home/adi/chevah/twisted/src/twisted/python/failure.py", line 481, in trap
    self.raiseException()
  File "/home/adi/chevah/twisted/src/twisted/python/failure.py", line 505, in raiseException
    raise self.value.with_traceback(self.tb)
twisted.internet.error.ProcessTerminated: A process has ended with a probable error condition: process ended with exit code 127.

twisted.python.test.test_util.GetPasswordTests.test_stdin
```

### PR DESCRIPTION
**Pull request by eevelweezel**
## Scope and purpose

Fixes #12060, #12099, #12062

## Contributor Checklist:

This process applies to *all* pull requests - no matter how small.
Have a look at [our developer documentation](https://docs.twisted.org/en/latest/core/development/dev-process.html) before submitting your Pull Request.

Below is a non-exhaustive list (as a reminder):

* The title of the PR should describe the changes and starts with the associated issue number, like “#1234 Brief description”.
* A release notes news fragment file was create in src/twisted/newsfragments/ (see: [Release notes fragments docs.](https://docs.twisted.org/en/latest/core/development/dev-process.html#release-notes-management))
* The automated tests were updated.
* Once all checks are green, request a review by leaving a comment that contains exactly the string `please review`.
  Our bot will trigger the review process, by applying the pending review label
  and requesting a review from the Twisted dev team.

### CODE REVIEW

**Comment by itamarst** [comment]
Going to finish this up.

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -137,18 +137,36 @@ jobs:
...
+            tox-env: 'alldeps-withcov-posix'
+            job-name: 'alldeps-3.13'
+
+          # Pre-release Python 3.13 on macOS with default settings.
+          - python-version: '3.13-dev'
+            runs-on: 'macos-12'
+            tox-env: 'macos-withcov-alldeps'
+            job-name: 'macos-alldeps-3.13'
+
+          # pywin32 doesn't have wheels for Python 3.13 yet.
```
Comment (adiroiban) [inline .github/workflows/test.yaml]: I think that we can keep this commented code.


But maybe move it after the other Window jobs.

It should make it easier to enable it later while we develop 3.13 support for Windows

Maybe add a link to the info about when the next pywin32 release will be available 

```suggestion
          # See https://github.com/mhammond/pywin32/issues/2294
          # pywin32 doesn't have wheels for Python 3.13 yet.
```

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -137,18 +137,36 @@ jobs:
...
-            # FIXME:https://github.com/twisted/twisted/issues/12060
-            # Also add twisted.python
-            # FIXME:https://github.com/twisted/twisted/issues/12061
-            # Add full twisted.test
-            trial-target: 'twisted.conch twisted.cred twisted.internet twisted.logger twisted.mail twisted.names twisted.pair twisted.persisted twisted.protocols twisted.runner twisted.spread twisted.test.test_defer twisted.trial twisted.web twisted.words'
-            # Disable concurrency for now, since we get occasional segfaults and
-            # this helps debug them:
-            trial-args: '--reactor=default'
+            tox-env: 'alldeps-withcov-posix'
+            job-name: 'alldeps-3.13'
```
Comment (adiroiban) [inline .github/workflows/test.yaml]: Do we need the extra tox-env and job name ?
Maybe we can go with the default settings.

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -137,18 +137,36 @@ jobs:
...
+
+          # Pre-release Python 3.13 on macOS with default settings.
+          - python-version: '3.13-dev'
+            runs-on: 'macos-12'
+            tox-env: 'macos-withcov-alldeps'
+            job-name: 'macos-alldeps-3.13'
```
Comment (adiroiban) [inline .github/workflows/test.yaml]: Maybe it's better to move them after `job-name: 'macos-12-default-tests'` 


For macos we run limtied tests... and I guess that once py3.13 is released we will stop running tests on macoS with py3.12

**Review by adiroiban — APPROVED** [review]
Excelent update. Thanks

I think that this can be merged.

Only minor comments regarding the GitHub Actions yaml organization.

Happy to see that we run more tests

-----

I see that some tests are skipped on 3.13 ... but I don't know how to check which one are skipped

* 3.12 tests on Linux PASSED (skips=503, successes=10970)

* 3.13 tests on Linux PASSED (skips=530, successes=10943)

maybe they are the `crypt` and `spwd` tests

**Pull request merged**